.github/workflows/haskell-ci.yml

```diff
@@ -6,79 +6,144 @@
 #
 #   haskell-ci regenerate
 #
-# For more information, see https://github.com/haskell-CI/haskell-ci
+# For more information, see https://github.com/andreasabel/haskell-ci
```

Comment:
Could you explain why we're switching to your own fork? Have to confess I'm a bit reluctant to, unless strictly necessarily. If it necessary, then documentation as to the reason should be added here.

Reply:
My fork uses the `ubuntu-20.04` image rather than the `ubuntu-18.04` image which allows it to use the latest version of `action/checkout`.  Otherwise, it is interchangeable, and you could as well regenerate the CI with the original `haskell-ci`.

Reply:
And why do we need the latest version of `action/checkout`?

Reply:
Not needed.  
But you might get harassed by dependabot if you don't use it...

If it matters to you, I can install the original haskell-ci and recreate this workflow.  But this is anyway just customary CI for 1.7.3 which won't end up in master.

Reply:
Yes, I guess that's my real question! Why are we trying to make a custom CI pass for v1.7.3?

Reply:
I don't want to release without having run CI.  But I now I am passing the bucket to you, so I make #2141 without CI changes.